src/examples/signal_generator.cpp

```diff
@@ -0,0 +1,239 @@
+          "frame_id: { protocol: 'api' },"
+          "fix_dynamic: { protocol: 'api' }"
+        "}"
+      "}");
+    ComponentConfiguration digital_conf = ComponentConfiguration("{"
+      "outputs: {"
+        "digital: { protocol: 'api' },"
+        "fix_dynamic: { protocol: 'api' }"
+      "},"
+      "inputs: {"
+        "value: { protocol: 'api' }"
+    "}}");
+
+    auto remote_endpoint = ConnectionConfiguration("{ value: { endpoint: '/signal/value' } }");
+
+  #endif
+
+  //-------------------------------------------------
+  // Step 3: Initialize state machines
+  //-------------------------------------------------
+
+  signal_sm.init(signal_conf);
+  digital_sm.init(digital_conf);
+
+  //-------------------------------------------------
+  // Step 4: Start state machines
+  //-------------------------------------------------
+
+  signal_sm.start();
+  digital_sm.start();
+
+  //-------------------------------------------------
+  // Step 5: Connect state machines
+  //-------------------------------------------------
+
+  signal_sm.connect();
+  digital_sm.connect(remote_endpoint);
+
+  //-------------------------------------------------
+  // Step 6: Check state machines
+  //-------------------------------------------------
+
+  signal_sm.check();
+  digital_sm.check();
+
+  //-------------------------------------------------
+  // Step 7: Update state machines
+  //-------------------------------------------------
+
+  signal_sm.update();
+  digital_sm.update();
+
+  //-------------------------------------------------
+  // Step 8: Create cancellation token and sleep for 
+  //         1s
+  //-------------------------------------------------
+
+  // This object is a single token, shared by all the threads, used to ask them to stop running.
+  CancellationTokenSource cancellation_token;
+  std::this_thread::sleep_for(1s);
+
+  //-------------------------------------------------
+  // Step 9: Create spinners
+  //-------------------------------------------------
+
+  StateMachineSpinner signal_spinner(signal_sm, cancellation_token, 10ms);
+  StateMachineSpinner digital_spinner(digital_sm, cancellation_token, 10ms);
+
+  //-------------------------------------------------
+  // Step 10: Create dynamic property access
+  //-------------------------------------------------
+
+  // Access the dynamic properties from the SignalGeneratorComponent
+  hyro::DynamicPropertyAccess dynamic_property_access_signal("/signal"_uri);
+  hyro::DynamicPropertyAccess dynamic_property_access_digital("/digital"_uri);
+
+  //-------------------------------------------------
+  // Step 11: Start plot function as thread
+  //-------------------------------------------------
+
+  std::thread t1(plot);
+  std::this_thread::sleep_for(2s);
+
+  //-------------------------------------------------
+  // Step 12: Request parameters from user and set
+  //          new dynamic properties
+  //-------------------------------------------------
+
+  while (true)
```

Comment:
You program will never end, just pressing`Ctrl+C` on the terminal.

I suggest you add a question **"Continue? (Y/N)"** after setting the dynamic parameters. If the user inform **"N"** stop the loop.

Other option is call the **plot()** in the main thread and move this while to a thread. In this way, you can exit the program when the user close all windows.